compiler/rustc_middle/src/query/mod.rs

```diff
@@ -1077,7 +1077,7 @@ rustc_queries! {
     }
 
     /// Return all `impl` blocks in the current crate.
-    query all_local_trait_impls(_: ()) -> &'tcx BTreeMap<DefId, Vec<LocalDefId>> {
+    query all_local_trait_impls(_: ()) -> &'tcx FxHashMap<DefId, Vec<LocalDefId>> {
```

Comment:
Is it possible iteration order matters here?

Reply:
Possibly... Here are the uses of `all_local_trait_impls` and `hir().trait_impls`:

`all_local_trait_impls`:

- [compiler/rustc_middle/src/hir/map/mod.rs#L552](https://github.com/pierwill/rust/blob/39fc8d73e8b7aa062d8da8f4fd0c7b70cae62d32/compiler/rustc_middle/src/hir/map/mod.rs#L552)
- [compiler/rustc_middle/src/hir/mod.rs#L109](https://github.com/pierwill/rust/blob/39fc8d73e8b7aa062d8da8f4fd0c7b70cae62d32/compiler/rustc_middle/src/hir/mod.rs#L109)
- [compiler/rustc_middle/src/query/mod.rs#L1080](https://github.com/pierwill/rust/blob/39fc8d73e8b7aa062d8da8f4fd0c7b70cae62d32/compiler/rustc_middle/src/query/mod.rs#L1080)
- [compiler/rustc_typeck/src/coherence/mod.rs#L203](https://github.com/pierwill/rust/blob/39fc8d73e8b7aa062d8da8f4fd0c7b70cae62d32/compiler/rustc_typeck/src/coherence/mod.rs#L203)
- [compiler/rustc_typeck/src/coherence/orphan.rs#L15](https://github.com/pierwill/rust/blob/39fc8d73e8b7aa062d8da8f4fd0c7b70cae62d32/compiler/rustc_typeck/src/coherence/orphan.rs#L15)

`hir().trait_impls`:

- [compiler/rustc_infer/src/infer/error_reporting/nice_region_error/static_impl_trait.rs#L446](https://github.com/pierwill/rust/blob/39fc8d73e8b7aa062d8da8f4fd0c7b70cae62d32/compiler/rustc_infer/src/infer/error_reporting/nice_region_error/static_impl_trait.rs#L446)
- [compiler/rustc_middle/src/ty/trait_def.rs#L222](https://github.com/pierwill/rust/blob/39fc8d73e8b7aa062d8da8f4fd0c7b70cae62d32/compiler/rustc_middle/src/ty/trait_def.rs#L222)
- [compiler/rustc_typeck/src/coherence/builtin.rs#L41](https://github.com/pierwill/rust/blob/39fc8d73e8b7aa062d8da8f4fd0c7b70cae62d32/compiler/rustc_typeck/src/coherence/builtin.rs#L41)
- [compiler/rustc_typeck/src/coherence/mod.rs#L189](https://github.com/pierwill/rust/blob/39fc8d73e8b7aa062d8da8f4fd0c7b70cae62d32/compiler/rustc_typeck/src/coherence/mod.rs#L189)

Reply:
If the iteration order matters, we may need to use an FxIndexMap and audit how these maps are built.